vault/resource_kv_secret_v2.go

```diff
@@ -108,6 +147,33 @@ func getKVV2Path(mount, name, prefix string) string {
 	return fmt.Sprintf("%s/%s/%s", mount, prefix, name)
 }
 
+func getKVV2MetadataPath(mount, name string) string {
+	return fmt.Sprintf("%s/metadata/%s", mount, name)
+}
+
+func getCustomMetadata(d *schema.ResourceData) map[string]interface{} {
+	data := map[string]interface{}{}
+
+	metadataFields := []string{
```

Comment:
nit: variable name is a bit verbose, relative to its function. May want to rename to `fields`